app/dbsetup.py

```diff
@@ -0,0 +1,68 @@
+from functools import partial, cached_property
+
+from flask_sqlalchemy import (BaseQuery, SignallingSession, SQLAlchemy, get_state)
+from sqlalchemy import orm
+from flask import _app_ctx_stack
+
+
+class ExplicitRoutingSession(SignallingSession):
+    """
+    This session implementation will route to explicitly named bind.
+    If no bind is mentioned with the session via the `using_bind` function,
+    then the `reader` bind will get returned instead.
+    """
+
+    _name = None
+
+    def get_bind(self, mapper=None, clause=None):
+        # If reader and writer binds are not configured,
+        # connect using the default SQLALCHEMY_DATABASE_URI
+        if not self.binds_setup:
+            return super().get_bind(mapper, clause)
+
+        return self.load_balance(mapper, clause)
+
+    def load_balance(self, mapper=None, clause=None):
+        # Use the explicit name if present
+        if self._name and not self._flushing:
+            bind = self._name
+            self._name = None
+            self.app.logger.debug(f"Connecting -> {bind}")
+            return get_state(self.app).db.get_engine(self.app, bind=bind)
+
+        # Everything else goes to the writer engine
+        else:
+            self.app.logger.debug("Connecting -> writer")
+            return get_state(self.app).db.get_engine(self.app, bind='writer')
+
+    def using_bind(self, name):
+        self._name = name
+        return self
+
+    @cached_property
+    def binds_setup(self):
+        binds = self.app.config['SQLALCHEMY_BINDS'] or {}
+        return all([k in binds for k in ['reader', 'writer']])
```

Comment:
I was thinking we could have constants for the `reader` and `writer` strings. Hence consumers of the API wouldn't make a mistake. We could even just have `using_master` and `using_reader` methods, knowing we won't have other binds in the near future anyway.